### PR TITLE
refactor(calcite-dropdown): miscellaneous changes in response to review

### PR DESCRIPTION
**Pull request by MikeTschudi**
**Related Issue:** PR #1386

## Summary
PR #1386 has several recommendations from @jcfranco's review; this PR implements them

### CODE REVIEW

File: src/components/calcite-dropdown/calcite-dropdown.e2e.ts [hunk 1]
```diff
@@ -857,39 +858,75 @@ describe("calcite-dropdown", () => {
...
-    <calcite-dropdown-item id="item-6">
-    Dropdown Item Content
-    </calcite-dropdown-item>
-    <calcite-dropdown-item id="item-7" href="google.com">
-    Dropdown Item Content
-    </calcite-dropdown-item>
-    </calcite-dropdown-group>
-    </calcite-dropdown>`);
+    await page.setContent(dedent`
+      <calcite-dropdown>
```
Comment (jcfranco) [inline src/components/calcite-dropdown/calcite-dropdown.e2e.ts:899]: Nitpick: if this HTML is the same as above, storing them in a variable to share would clean up things a bit.

**Review by jcfranco — APPROVED** [review]
Awesome. Thanks, @MikeTschudi! 💥

**Comment by macandcheese** [comment]
@MikeTschudi could you merge master in to give Travis another shot at this?

**Comment by eriklharper** [comment]
Looks like we should update the date story to hard-code a default date value so that the story doesn't have to change appearance every month when the current `Date()` changes.
image

**Review by paulcpederson — APPROVED** [review]

File: src/components/calcite-dropdown/calcite-dropdown.e2e.ts [hunk 2]
```diff
@@ -892,8 +892,7 @@ describe("calcite-dropdown", () => {
   });
 
   it("is accessible", async () => {
-    const page = await simplePageSetup(componentTagOrHTML);
-    accessible(dedent`${dropdownSelectionModeContent}`, page);
+    accessible(dedent`${dropdownSelectionModeContent}`);
```
Comment (driskull) [inline src/components/calcite-dropdown/calcite-dropdown.e2e.ts:895]: should this be `await accessible(..`?

Comment (MikeTschudi) [reply src/components/calcite-dropdown/calcite-dropdown.e2e.ts]: Thanks, @driskull--that did it!

**Pull request merged**